lib/community/widgets/post_card_list.dart

```diff
@@ -359,15 +360,25 @@ class _PostCardListState extends State<PostCardList> with TickerProviderStateMix
 
   Future<void> dismissRead() async {
     if (widget.postViews != null) {
-      setState(() {
-        showRead = false;
-      });
-      await Future.delayed(const Duration(milliseconds: 400));
+      if (!(widget.postViews!.length > 15)) {
+        widget.onScrollEndReached();
+      }
+      for (var post in widget.postViews!) {
+        if (post.postView.read) {
+          setState(() {
+            toRemoveSet.add(post.postView.post.id);
+          });
+          await Future.delayed(const Duration(milliseconds: 60));
+        }
+      }
+      await Future.delayed(const Duration(milliseconds: 800));
```

Comment:
What is this for again? Is it to wait for all the animations to finish before resetting the state?

Reply:
Yes. For some reason it does not wait for the loop to finish. There's probably a better way to do this, but I didn't figure one out yet.

Reply:
And the 60 ms wait is what causes the dismissals to be staggered.

Reply:
Ahh okay okay, sounds good. I do think theres a better way but we can leave it as it is for now!